clients/lsp-zig.el

```diff
@@ -0,0 +1,36 @@
+;;; lsp-zig.el --- lsp-mode Zig integration -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2021 Riccardo Binetti
+
+;; Author: Riccardo Binetti <rbino@gmx.com>
+;; Keywords: languages,tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;;  client for zls, the Zig language server
+
+;;; Code:
+
+(require 'lsp-mode)
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection "zls")
+  :major-modes '(zig-mode)
```

Comment:
We're in transition from `:major-mode` to better `lsp-activate-on`, so you may want to change this like as 
```suggestion
  :activation-fn (lsp-activate-on "zig")
```

Reply:
@yyoncho Should we mention this in the docs when creating new client?

Reply:
@kiennq thanks for the heads-up, fixed it

Reply:
@kiennq yes, we should.